src/reddit_analysis.py

```diff
@@ -0,0 +1,48 @@
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
+
+
+class RedditAnalysis:
+    def __init__(self) -> None:
+        self.results = {}
+        self.stock_count = {}
+        
+    def run_analysis(self, dict):
+        analyzer = SentimentIntensityAnalyzer()
+
+        for sentence in dict:
+            self.sentiment_analyzer_scores(dict[sentence], analyzer)
+        
+        for key in dict.keys():
+            test = dict.get(key)
+            try:
+                self.stock_count[test['name']] = self.stock_count[test['name']] + 1
+            except KeyError:
+                self.stock_count[test['name']] = 1
+
+        for key, value in self.results.items():
+            value = value / self.stock_count.get(key)
+            self.results[key] = value
+
+        print(self.results)
+        return self.results
+
+    def sentiment_analyzer_scores(self, values, analyzer: SentimentIntensityAnalyzer):
```

Comment:
What if instead of passing the entire dictionary as an argument, you only pass the name and text u are using?